apisec/internal/timed/lru.go

```diff
@@ -45,17 +47,18 @@ type LRU struct {
 }
 
 // NewSet initializes a new, empty [LRU] with the given interval and clock
-// function. The provided interval must be at least 1 second, and may not exceed
-// [config.Interval].
+// function. A warning will be logged if it is set below 1 second. Panics if
+// the interval is more than [math.MaxUint32] seconds, as this value cannot be
+// used internally.
 //
 // Note: timestamps are stored at second resolution, so the interval will be
 // rounded down to the nearest second.
 func NewSet(interval time.Duration, clock ClockFunc) *LRU {
```

Comment:
side comment: function name is a little confusing after the renaming

Reply:
Good catch! I just don't see the name anymore at this point -_-